pysb/core.py

```diff
@@ -1426,8 +1427,15 @@ def enable_synth_deg(self):
             self.add_component(Parameter('__source_0', 1.0, _export=False))
 
         source_cp = as_complex_pattern(self.monomers['__source']())
-        if not any(source_cp.is_equivalent_to(other_cp) for other_cp, value in self.initial_conditions):
-            self.initial(source_cp, self.parameters['__source_0'])
+        if self.compartments:
+            for c in self.compartments:
+                source_cp.compartment = c
```

Comment:
This is a bug. Because it modifies source_cp in place, on the second loop it ends up altering the initial condition added in the first loop, making that one match the current proposed source_cp, and skipping it. This keeps happening until the end of the loop, at which point there's only one \__source initial condition for the last compartment in the model. source_cp needs to be copied on each loop, and the easiest way is just to use the proper `**` operator to refine it against the current compartment. Just replace this line with `source_cp = source_cp ** c`.